File: Checklists_Code/Checklists_Code/Checklist.m
//
//  Checklist.m
//  Checklists_Code
//
//  Created by wangx on 15/6/24.
//  Copyright (c) 2015年 wxiang1991. All rights reserved.
//

#import "Checklist.h"
#import "ChecklistItem.h"
@implementation Checklist

-(id)initWithCoder:(NSCoder *)aDecoder{
    if((self =[super init])){
        self.name = [aDecoder decodeObjectForKey:@"Name"];
        self.items = [aDecoder decodeObjectForKey:@"Items"];
        self.iconName = [aDecoder decodeObjectForKey:@"IconName"];
    }
    return self;
}

-(void)encodeWithCoder:(NSCoder *)aCoder{
    [aCoder encodeObject:self.name forKey:@"Name"];
    [aCoder encodeObject:self.items forKey:@"Items"];
    [aCoder encodeObject:self.iconName forKey:@"IconName"];
}

//计算没有完成的代办事项
-(int)countUncheckedItems{
    int count = 0;
    for(ChecklistItem *item in self.items){
        if(!item.checked){
            count+=1;
        }
    }
    return count;
}

@end

File: ChecklistSwift/ChecklistSwift/Checklist.swift
//
//  Checklist.swift
//  ChecklistSwift
//
//  Created by 王享 on 15/12/25.
//  Copyright © 2015年 王享. All rights reserved.
//

import UIKit

class Checklist: NSObject,NSCoding{
    
    var name:String?
    
    var iconName:String?
    
    var items:Array<ChecklistItem>?
    
    override init(){
        super.init();
    }
    
    required init?(coder aDecoder: NSCoder){
        self.name = aDecoder.decodeObject(forKey: "Name") as? String
        self.iconName = aDecoder.decodeObject(forKey: "IconName") as? String
        self.items = aDecoder.decodeObject(forKey: "Items") as? Array
    }
    
    func encode(with aCoder: NSCoder) {
        aCoder.encode(self.name, forKey: "Name")
        aCoder.encode(self.iconName, forKey: "IconName")
        aCoder.encode(self.items, forKey: "Items")
    }
    
    
    func countUncheckedItems() -> Int{
        var count:Int = 0
        if self.items != nil {
            for item:ChecklistItem in self.items! {
                if !item.checked! {
                    count += 1
                }
            }
        }
        return count
    }
}
